integreat_cms/cms/views/utils/search_content_ajax.py

```diff
@@ -98,21 +100,35 @@ def search_content_ajax(request, region_slug=None, language_slug=None):
 
     if user.has_perm("cms.view_page") and "page" in object_types:
         object_types.remove("page")
-        # Here a filter is not possible since archived is a property on PageTranslation
-        page_translations = (
-            page_translation
-            for page_translation in PageTranslation.search(region, language_slug, query)
-            if page_translation.page.archived == archived_flag
-            and page_translation.status == status.PUBLIC
-        )
-        results.extend(
-            format_object_translation(obj, "page") for obj in page_translations
+        pages = {
+            page.id: page
+            for page in region.pages.filter(
+                explicitly_archived=archived_flag
+            ).cache_tree()[0]
+        }
+        query = (
+            PageTranslation.search(region, language_slug, query)
+            .filter(page__in=pages.values(), status=status.PUBLIC)
+            .select_related("page__region", "language")
         )
 
+        for page_translation in query:
+            # Copy the cached attributes from the ``pages`` query to this page translation to prevent database queries
+            # pylint: disable=protected-access
+            cached_page = pages[page_translation.page.id]
+            page_translation.page._cached_ancestors = cached_page._cached_ancestors
+            page_translation.page.prefetched_public_translations = (
+                cached_page.prefetched_public_translations
+            )
+
+            results.append(format_object_translation(page_translation, "page"))
```

Comment:
```suggestion
        pages = region.pages.all().cache_tree(archived=archived_flag)[0]
        for page in pages:
            page_translation = page.get_translation(language_slug)
            if page_translation and (
                query.lower() in page_translation.slug
                or query.lower() in page_translation.title.lower()
            ):
                results.append(format_object_translation(page_translation, "page"))
```
Since we already fetched all pages including their translations from the database, we can also do the search in Python instead of another database query.